enhancements/network/egress-ip-per-destinations.md

```diff
@@ -0,0 +1,327 @@
+The two `EgressIP`s selects the same set of pods but different `TrafficSelector`s to select the previously created `EgressIPTraffic` CRs.
+The `EgressIP`s are automatically assigned to the secondary host interfaces on the egress node.
+
+The cluster admin gives permissions to the **application admin** to the namespaces.
+
+The **application admin** deploys its application and the application has a consistent but different source IP for the
+health signal traffic and the work load traffic.
+
+### API Extensions
+
+Add `TrafficSelector` field to `EgressIPSpec`:
+
+```golang
+// EgressIPSpec is a desired state description of EgressIP.
+type EgressIPSpec struct {
+	// EgressIPs is the list of egress IP addresses requested. Can be IPv4 and/or IPv6.
+	// This field is mandatory.
+	EgressIPs []string `json:"egressIPs"`
+	// TrafficSelector applies the egress IP only to the network traffic defined within the selected EgressIPTraffic(s).
+	// If not set, all egress traffic is selected.
+	// +optional
+	TrafficSelector metav1.LabelSelector `json:"trafficSelector"`
+	// NamespaceSelector applies the egress IP only to the namespace(s) whose label
+	// matches this definition. This field is mandatory.
+	NamespaceSelector metav1.LabelSelector `json:"namespaceSelector"`
+	// PodSelector applies the egress IP only to the pods whose label
+	// matches this definition. This field is optional, and in case it is not set:
+	// results in the egress IP being applied to all pods in the namespace(s)
+	// matched by the NamespaceSelector. In case it is set: is intersected with
+	// the NamespaceSelector, thus applying the egress IP to the pods
+	// (in the namespace(s) already matched by the NamespaceSelector) which
+	// match this pod selector.
+	// +optional
+	PodSelector metav1.LabelSelector `json:"podSelector,omitempty"`
+}
+```
+
+New CRD:
+
+```golang
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:nostatus
+// +kubebuilder:resource:shortName=eipd,scope=Cluster
+// +kubebuilder:printcolumn:name="DestinationNetworks",type=string,JSONPath=".spec.destinationNetworks"
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// EgressIPTraffic defines a set of networks outside the cluster network.
+type EgressIPTraffic struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec EgressIPTrafficSpec `json:"spec"`
+}
+
+// EgressIPTrafficSpec defines the desired state description of EgressIPTraffic.
+// +kubebuilder:validation:MaxProperties=1
+// +kubebuilder:validation:MinProperties=1
+type EgressIPTrafficSpec struct {
+	// DestinationNetworks is the list of Network CIDRs that defines external destinations
+	// for IPv4 or IPv6. Mixing IP families are not allowed.
```

Comment:
how are you preventing this

Reply:
What about mixing IPv4 and IPv6 by havving an EgressIPTraffic CR for each (one IPv4 and one IPv6) with the same label. The EIP would then bind both of them. Would this be supported? In the near-term use cases dual stack clusters are applicable.

Reply:
After talking with the team, its not good practise to prevent the mixing of IP families. Removed this requirement.